lib/Form/Util/CustomFonts.php

```diff
@@ -87,12 +87,26 @@ public function enqueueStyle() {
       // When we load all custom fonts in one request, a form from WC Payments isn't displayed correctly.
       // It looks that the larger file size overloads the Stripe SDK.
       foreach (array_chunk(self::FONTS, self::FONT_CHUNK_SIZE) as $key => $fonts) {
-        $this->wp->wpEnqueueStyle('mailpoet_custom_fonts_' . $key, $this->generateLink($fonts));
```

Comment:
@wxa I totally understand. I wanted to reuse the generateLink function without having to create a new instance of the class, hence the static here.

My idea behind choosing the static route instead of the instance route is due to the function being a pure function. It requires no additional dependency to use. I believe it's actually much earlier to test as well.

This was supposed to be used here: `lib/Form/Widget.php` but unfortunately an instance of `lib/Form/Util/CustomFonts.php` class does not exist at that point of Widget class initialization on the ContainerWrapper::getInstance()-get method. 

I would have had to create a new instance of CustomFonts class providing all the additional dependencies. This is not really scalable as it is not making use of the Dependency injection mechanism and adding more dependencies to the CustomFonts class in the future will break the Widget class.

These were my reasoning for using the static function.

Reply:
Thanks for explaining. 

>  unfortunately an instance of lib/Form/Util/CustomFonts.php class does not exist at that point of Widget class initialization on the ContainerWrapper::getInstance()-get method.

It's because the `Widget` class itself does not use dependency injection for some reason. If you set the `CustomFonts` class as public in `lib/DI/ContainerConfigurator`, it will be possible to explicitly get an instance of it in `Widget` via the container. Then the reasoning for not using an instance because of an overhead becomes irrelevant.

> My idea behind choosing the static route instead of the instance route is due to the function being a pure function. It requires no additional dependency to use. I believe it's actually much earlier to test as well.

Even if a static method has no dependencies, it's harder to stub it out when it's used in another class. I thought we had a coding rule for that, but I can't find it, so I will ask the team in Slack.

Reply:
Update: as [discussed in Slack](https://a8c.slack.com/archives/C01G2EDTMQF/p1637136285030500), preferring instance methods over static ones looks to be a rule we missed in our coding guidelines. We'll update them.

Reply:
@wxa 
Noted.
I will update the PR.